pkg/tfgen/installation_docs.go

```diff
@@ -124,3 +144,192 @@ func applyEditRules(contentBytes []byte, docFile *DocFile) ([]byte, error) {
+			chooserEnd := "{{< /chooser >}}\n"
+
+			// Generate each language in turn, so that we can mark up the output with the correct Hugo shortcodes.
+			// TODO: we want to use pulumi code choosers in the future, but resourcedocsgen does not support this yet.
+			for _, lang := range langs {
+				langSlice := []string{lang}
+				choosableStart := fmt.Sprintf("{{%% choosable language %s %%}}\n", lang)
+				choosableEnd := "\n{{% /choosable %}}\n"
+
+				// Generate the Pulumi.yaml config file for each language
+				configFile, err := writeConfigYamlViaCLI(outDir, lang, mappingsArgs)
+				if err != nil {
+					return "", err
+				}
+				// Generate example itself
+				convertedLang, err := g.convertHCL(conversionResult, code, exPath.String(), langSlice)
+				if err != nil {
+					return "", err
+				}
+				returnContent = returnContent + choosableStart + configFile + convertedLang + choosableEnd
+			}
+			returnContent += chooserEnd
+			startIndex = block.end + len(codeFence)
+		} else {
+			// Write any code block as-is.
+			returnContent = returnContent + contentStr[block.start:block.end+len(codeFence)]
+			startIndex = block.end + len(codeFence)
+		}
+	}
+	// Write any remainder.
+	returnContent = returnContent + contentStr[codeBlocks[len(codeBlocks)-1].end+len(codeFence):]
+	return returnContent, cleanupErr
+}
+
+// This function obtains the Pulumi.yaml config file for a given language.
+// It relies on the presence of a `main.tf` file for conversion.
+func writeConfigYamlViaCLI(outDir, lang string, mappingsArgs []string) (string, error) {
+	// Check if there's a main.tf file in the current working directory (outDir)
+	if _, err := os.Stat(filepath.Join(outDir, "main.tf")); err != nil {
+		return "", fmt.Errorf("expected main.tf file %w", err)
+	}
+	// Get the pulumi command
+	pulumiPath, err := exec.LookPath("pulumi")
+	if err != nil {
+		return "", fmt.Errorf("couldn't find pulumi path")
+	}
+
+	cmdArgs := []string{
+		"convert",
+		"--from",
+		"terraform",
+		"--language",
+		lang,
+		"--generate-only",
+	}
+	cmdArgs = append(cmdArgs, mappingsArgs...)
+
+	cmd := exec.Command(pulumiPath, cmdArgs...)
+	cmd.Dir = outDir
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout, cmd.Stderr = &stdout, &stderr
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("convertViaPulumiCLI: pulumi command failed: %w\n"+
+			"Stdout:\n%s\n\n"+
+			"Stderr:\n%s\n\n",
+			err, stdout.String(), stderr.String())
+	}
+	pulumiYAMLFileBytes, err := os.ReadFile(filepath.Join(outDir, "Pulumi.yaml"))
+	if err != nil {
+		return "", err
+	}
+	// Remove the temp directory's random number from project name
+	regex := regexp.MustCompile(`configuration-example.*`)
+	pulumiYAMLFileBytes = regex.ReplaceAll(pulumiYAMLFileBytes, []byte("configuration-example"))
+
+	// Add descriptive code comment
+	configFile := "```yaml\n" +
+		"# Pulumi.yaml provider configuration file\n" +
+		string(pulumiYAMLFileBytes) + "\n```\n"
+	return configFile, nil
```

Comment:
Are these configuration files ever different from non-primitive files? Can we just embed Pulumi.yaml per supported language? They look really primitive and there's not a lot going on in the generator in https://github.com/pulumi/pulumi-converter-terraform/blob/7879de9e9ac884ea9575d8c47866bcb2d6c26161/pkg/convert/tf.go#L2898